cleanlab/dataset.py

```diff
@@ -0,0 +1,416 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+
+"""Dataset Module
+Supports dataset-level and class-level automated quality, including finding which classes should
+be merged with other classes in your dataset, which classes should be removed,
+and which classes tend to be annotated most/least correctly overall.
+"""
+
+import numpy as np
+import pandas as pd
+from cleanlab.count import estimate_joint
+
+
+def rank_classes_by_label_quality(
+    labels=None,
+    pred_probs=None,
+    *,
+    class_names=None,
+    num_examples=None,
+    joint=None,
+    confident_joint=None,
+    multi_label=False,
+):
+    """Returns a Pandas DataFrame with all classes and three overall class label quality scores
+     (details about each score are listed in the Returns parameter). By default, classes are ordered
+     by "Overall Label Noise Score" descending.
+
+    Score values are unnormalized and may tend to be very small. What matters is their relative
+    ranking across the classes.
+
+    Parameters
+    ----------
+    For parameter info, see the docstring of `dataset.find_overlapping_classes`
+
+    Returns
+    -------
+    pd.DataFrame
+        A Pandas DataFrame with cols "Class Index", "Overall Label Issues", "Inverse Label Issues",
+        "Overall Label Issues", "Inverse Label Noise Score", "Overall Label Quality Score" and a
+        description of each below. Noise scores are between 0 and 1, where 0 implies no label issues
+        in the class. The "Overall Label Quality Score" is also between 0 and 1 where 1 implies
+        perfect quality.
+        Columns:
+        * "Class Index"
+            The index of the class in 0, 1, ..., num_classes - 1.
+        * Overall Label Issues
+            count(given_label = k, true_label != k)
+            Estimated number of label issues in the class (usually the most accurate method).
+        * Inverse Label Issues
+            count(given_label != k, true_label = k)
+            Estimated number of times this class occurs in other classes in the dataset.
+        * Overall Label Noise Score
+            prob(true_label != k | given_label = k)
+            Estimated proportion of label issues in the class.
+        * Inverse Label Noise Score
+            prob(given_label != k | true_label = k)
+            Estimated proportion of times this class label occurs in other classes in the dataset.
+        * Overall Label Quality Score
+            p(given_label = k | true_label = k)
+            How accurate the labels are in each class as a conditional prob p(given=k | true=k).
+        By default, the DataFrame is ordered by "Overall Label Noise Score" descending.
```

Comment:
Shouldn't the whole cleanlab package stick with the convention:
 lower scores -> more serious problem that warrants attention
 like we did for our current (per-example) label-quality scores?
I think we need to stick with one convention throughout, otherwise user has to remember what higher/lower means for every score and we'll soon have tons of scores.

Reply:
Removed "Score", so now it's just "Overall Label Noise"